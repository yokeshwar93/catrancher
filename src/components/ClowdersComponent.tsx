import React from 'react';
import {Grid, makeStyles} from "@material-ui/core";
import Paper from "@material-ui/core/Paper";
import {clowderProps} from "../models/modelTypes";


const useStyles = makeStyles((theme) => ({
    root: {
        flexGrow: 1,
        marginTop: '1%'
    },
    paper: {
        height: '10vh',
        width: '5vw',
        border: '1px solid #0593cf',
        borderRadius: '0px'
    },
    imageStyle: {
        height: '95%',
        width: '95%',
        marginLeft: 'auto',
        marginRight: 'auto',
        display: 'block'
    }

}));

const ClowdersComponent = (props : clowderProps) => {
    const classes = useStyles();
    const { clowders } = props;

    return (

        <Grid container className={classes.root} spacing={2}>
            <Grid item xs={12}>
                {[1, 2, 3, 4].map((index : any) => (
                    <Grid container  key={index} justifyContent="center" spacing={0}>
                        {[0, 1, 2].map((value) => (
                            <Grid key={value} item>
                                <Paper className={classes.paper} elevation={0} >
                                    {clowders[index][value] &&
                                        <img className={classes.imageStyle}
                                             src={clowders[index][value].imageUrl}
                                             alt={`Cat image_${clowders[index][value].id}`}/>
                                    }
                                </Paper>
                            </Grid>
                        ))}
                    </Grid>
                ))}
            </Grid>
        </Grid>
    )
}

export default ClowdersComponent;