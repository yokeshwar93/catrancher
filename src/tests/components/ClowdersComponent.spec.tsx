import React from 'react';
import { render } from '@testing-library/react';
import { mockStore } from "../store/reducer/mockData";
import ClowdersComponent from "../../components/ClowdersComponent";

describe('renders selected clowder', () => {

    it('render second grid with valid clowders', () => {
        const { container } = render(
            <ClowdersComponent clowders={mockStore.clowders}  />
        );
        expect(container).toMatchSnapshot();

    })

});
