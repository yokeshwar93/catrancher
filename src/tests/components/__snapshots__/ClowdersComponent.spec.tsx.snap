// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders selected clowder render second grid with valid clowders 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation0 MuiPaper-rounded"
          >
            <img
              alt="Cat image_1ttg"
              class="makeStyles-imageStyle-3"
              src="Image url"
            />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation0 MuiPaper-rounded"
          >
            <img
              alt="Cat image_2wsb"
              class="makeStyles-imageStyle-3"
              src="Image url"
            />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation0 MuiPaper-rounded"
          >
            <img
              alt="Cat image_3brr"
              class="makeStyles-imageStyle-3"
              src="Image url"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation0 MuiPaper-rounded"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation0 MuiPaper-rounded"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation0 MuiPaper-rounded"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation0 MuiPaper-rounded"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation0 MuiPaper-rounded"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation0 MuiPaper-rounded"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation0 MuiPaper-rounded"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation0 MuiPaper-rounded"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation0 MuiPaper-rounded"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
